drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/editor/menu/AbstractColumnMenuPresenterTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.scenariosimulation.client.editor.menu;
+
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.drools.workbench.screens.scenariosimulation.client.events.DeleteColumnEvent;
+import org.drools.workbench.screens.scenariosimulation.client.events.DuplicateInstanceEvent;
+import org.drools.workbench.screens.scenariosimulation.client.events.InsertColumnEvent;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.verify;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class AbstractColumnMenuPresenterTest extends AbstractMenuTest {
+
+    @Before
+    public void setup() {
+        abstractColumnMenuPresenter = new AbstractColumnMenuPresenter() {
+
+            @Override
+            public void callSuperShow(int mx, int my) {
+            }
+        };
+        super.setup();
+    }
+
+    @Test
+    public void initMenu() {
+        assertNull(abstractColumnMenuPresenter.columnContextLIElement);
+        assertNull(abstractColumnMenuPresenter.insertColumnLeftLIElement);
+        assertNull(abstractColumnMenuPresenter.insertColumnRightLIElement);
+        assertNull(abstractColumnMenuPresenter.deleteColumnInstanceLIElement);
+        assertNull(abstractColumnMenuPresenter.duplicateInstanceLIElement);
+        abstractColumnMenuPresenter.initMenu();
+        assertNotNull(abstractColumnMenuPresenter.columnContextLIElement);
+        assertNotNull(abstractColumnMenuPresenter.insertColumnLeftLIElement);
+        assertNotNull(abstractColumnMenuPresenter.insertColumnRightLIElement);
+        assertNotNull(abstractColumnMenuPresenter.deleteColumnInstanceLIElement);
+        assertNotNull(abstractColumnMenuPresenter.duplicateInstanceLIElement);
+    }
+
+    @Test
+    public void show() {
+        commonShow(4, 5, false, false);
+        commonShow(4, 5, false, true);
+        commonShow(4, 5, true, false);
+        commonShow(4, 5, true, true);
+    }
+
+    private void commonShow( int mx, int my, boolean asProperty, boolean showDuplicateInstance) {
```

Comment:
If we call `reset(abstractColumnMenuPresenterSpy)` as last command of the `commonShow`, then we can change all `verify(abstractColumnMenuPresenterSpy, atLeastOnce())` to `verify(abstractColumnMenuPresenterSpy, times(1))` what is more precise test, WDYT please?

Reply:
@jomarko 
I've put `atLEastOnce()` instead of times(1)  because for test purpose I create only *one* `menuItemMock` and one `executableMenuItemMock` - so for every invocation where one of the two is used, the condition is verified.

Reply:
@gitgabrio I totally understand why you written in the current way. that I say is, there is option for more strict test, using the Mockito `reset` method. My question is if you considered it.

Reply:
@jomarko 
Updated with last commit